modules/core/src/main/scala/com.snowplowanalytics.iglu/client/resolver/registries/JavaNetLookup.scala

```diff
@@ -0,0 +1,142 @@
+          case Registry.Http(_, connection) => httpLookup(connection, schemaKey)
+          case Registry.Embedded(_, path)   => RegistryLookup.embeddedLookup[F](path, schemaKey)
+          case Registry.InMemory(_, schemas) =>
+            F.delay(RegistryLookup.inMemoryLookup(schemas, schemaKey))
+        }
+
+      def list(
+        registry: Registry,
+        vendor: String,
+        name: String,
+        model: Int
+      ): F[Either[RegistryError, SchemaList]] =
+        registry match {
+          case Registry.Http(_, connection) => httpList(connection, vendor, name, model)
+          case Registry.Embedded(_, base) =>
+            val path = RegistryLookup.toSubpath(base, vendor, name)
+            Sync[F].delay(Utils.unsafeEmbeddedList(path, model))
+          case _ => F.pure(RegistryError.NotFound.asLeft)
+        }
+    }
+
+  // Id instance also swallows all exceptions into `RegistryError`
+  implicit def idLookupInstance: RegistryLookup[Id] =
+    new RegistryLookup[Id] {
+      def lookup(repositoryRef: Registry, schemaKey: SchemaKey): Id[Either[RegistryError, Json]] =
+        repositoryRef match {
+          case Registry.Http(_, connection) =>
+            Utils
+              .stringToUri(RegistryLookup.toPath(connection.uri.toString, schemaKey))
+              .flatMap(uri => Utils.unsafeGetFromUri(uri, connection.apikey))
+          case Registry.Embedded(_, base) =>
+            val path = RegistryLookup.toPath(base, schemaKey)
+            Utils.unsafeEmbeddedLookup(path)
+          case Registry.InMemory(_, schemas) =>
+            RegistryLookup.inMemoryLookup(schemas, schemaKey)
+        }
+
+      def list(
+        registry: Registry,
+        vendor: String,
+        name: String,
+        model: Int
+      ): Id[Either[RegistryError, SchemaList]] =
+        registry match {
+          case Registry.Http(_, connection) =>
+            val subpath = RegistryLookup.toSubpath(connection.uri.toString, vendor, name, model)
+            Utils.stringToUri(subpath).flatMap(Utils.unsafeHttpList(_, connection.apikey))
+          case Registry.Embedded(_, base) =>
+            val path = RegistryLookup.toSubpath(base, vendor, name)
+            Utils.unsafeEmbeddedList(path, model)
+          case _ =>
+            RegistryError.NotFound.asLeft
+        }
+    }
+
+  /**
+   * Retrieves an Iglu Schema from the HTTP Iglu Repo as a JSON
+   *
+   * @param http endpoint and optional apikey
+   * @param key The SchemaKey uniquely identifying the schema in Iglu
+   * @return either a `Json` on success, or `RegistryError` in case of any failure
+   *         (i.e. all exceptions should be swallowed by `RegistryError`)
+   */
+  private[registries] def httpLookup[F[_]: Sync](
+    http: Registry.HttpConnection,
+    key: SchemaKey
+  ): F[Either[RegistryError, Json]] =
+    Utils
+      .stringToUri(RegistryLookup.toPath(http.uri.toString, key))
+      .traverse(uri => Utils.getFromUri(uri, http.apikey))
```

Comment:
It looks like `Utils` contains some `java.net...` related logic, could move such stuff from `Utils` here as well?